service/src/test/java/de/adorsys/psd2/sandbox/xs2a/pis/PaymentInitiationWithScaSteps.java

```diff
@@ -33,63 +38,92 @@
 
   private Context context = new Context();
 
-  @Given("^PSU initiated a single payment using the payment product (.*)$")
-  public void initiatePayment(String paymentProduct) {
-    context.setPaymentService("payments");
+  @Given("^PSU initiated a (.*) payment using the payment product (.*)$")
+  public void initiatePayment(String paymentType, String paymentProduct) {
     context.setPaymentProduct(paymentProduct);
 
+    Request request = null;
     HashMap<String, String> headers = TestUtils.createSession();
-    headers.put("psu-ip-address", "192.168.0.26");
+
+    if(paymentType.equals("single")){
+      request = getSinglePayment(headers, false);
+    }
+    if(paymentType.equals("future-dated")){
+      request = getSinglePayment(headers, true);
+    }
+    if(paymentType.equals("periodic")){
+      request = getPeriodicPayment(headers);
+    }
+
+    ResponseEntity<PaymentInitationRequestResponse201> response = template.exchange(
+        context.getPaymentService() + "/" +
+            context.getPaymentProduct(),
+        HttpMethod.POST,
+        request.toHttpEntity(),
+        PaymentInitationRequestResponse201.class);
+
+    context.setPaymentId(response.getBody().getPaymentId());
+  }
+  private Request<PaymentInitiationSctJson> getSinglePayment(HashMap<String, String> headers, boolean isFutureDated) {
+    context.setPaymentService("payments");
 
     PaymentInitiationSctJson payment = new PaymentInitiationSctJson();
     payment.setEndToEndIdentification("WBG-123456789");
-
-    HashMap<String, String> debtorAccount = new HashMap<>();
-    debtorAccount.put("currency", "EUR");
-    debtorAccount.put("iban", "DE51250400903312345678");
-    payment.setDebtorAccount(debtorAccount);
+    payment.setDebtorAccount(createAccount("DE51250400903312345678", "EUR"));
 
     Amount instructedAmount = new Amount();
     instructedAmount.setAmount("520");
     instructedAmount.setCurrency("EUR");
     payment.setInstructedAmount(instructedAmount);
 
-    HashMap<String, String> creditorAccount = new HashMap<>();
-    creditorAccount.put("currency", "EUR");
-    creditorAccount.put("iban", "DE15500105172295759744");
-    payment.setCreditorAccount(creditorAccount);
-
-    payment.setCreditorAgent("AAAADEBBXXX");
+    payment.setCreditorAccount(createAccount("DE15500105172295759744", "EUR"));
     payment.setCreditorName("WBG");
-
-    Address creditorAddress = new Address();
-    creditorAddress.setBuildingNumber("56");
-    creditorAddress.setCity("Nürnberg");
-    creditorAddress.setCountry("DE");
-    creditorAddress.setPostalCode("90543");
-    creditorAddress.setStreet("WBG Straße");
-    payment.setCreditorAddress(creditorAddress);
-
+    payment.setCreditorAddress(createCreditorAddress());
     payment.setRemittanceInformationUnstructured("Ref. Number WBG-1222");
+    if(isFutureDated){
+      payment.setRequestedExecutionDate(LocalDate.now());
```

Comment:
Shouldnt we at least use a date in the future for the future dated payment test?

Reply:
done